modules/market/src/main/java/com/opengamma/strata/market/curve/CurveInfoType.java

```diff
@@ -38,11 +38,11 @@
   /**
    * Obtains an instance from the specified name.
    * <p>
-   * Curve names may contain any character, but must not be empty.
+   * The name may contain any character, but must not be empty.
    *
-  * @param <T>  the type associated with the info
-   * @param name  the name of the curve
-   * @return a curve with the specified name
+   * @param <T>  the type associated with the info
+   * @param name  the name
+   * @return a type instance with the specified name
    */
   @FromString
   public static <T> CurveInfoType<T> of(String name) {
```

Comment:
This is a Joda-Convert type, so it cannot contain `Class` without much more complexity that simply isn't justified.

The type is captured and used in the constants at the top of the file, which are used in other parts of the system to make queries more usable. The alternative is a String to String map which is less safe than this design (where only comments were changed).